### ISSUE TITLE
Trigger UnboundLocalError while raise HttpsError

### ISSUE THREAD

**Issue by DinhHuy2010** [ThioJoe/YT-Spammer-Purge#728]
### Duplicate Issues

- [X] There are **no existing posts relating to my problem**
- [ ] There are **existing posts relating to my problem, but the solution given, doesn't work for me.**

### What happened?

Well, I was trying to hold for review I get some error from HttpError and UnboundLocalError at same time
HttpError reason: unexpectedParameter

### Release version

YTSpammerPurge 2.16.4

### Steps to reproduce

1. Scan recent video for other channel (Channel ID: UCTFBrIlufYVR80kMyIj-XUw)
2. Use Sensitive-Smart Mode as Moderator Mode
3. Hold for review
4. Get error

### What platform are you seeing this problem on?

Windows (.py script)

### Relevant log output

```shell
Traceback (most recent call last):
  File "C:\python\spam_base\YT-Spammer-Purge-2.16.4\Scripts\operations.py", line 1132, in check_deleted_comments
    results = auth.YOUTUBE.comments().list(
  File "C:\Users\huy04\AppData\Local\Programs\Python\Python39\lib\site-packages\googleapiclient\_helpers.py", line 131, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "C:\Users\huy04\AppData\Local\Programs\Python\Python39\lib\site-packages\googleapiclient\http.py", line 937, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments?part=snippet&id=UgyM8rTxfyeLEWwj5Gp4AaABAg.9ZC12Lvyc8a9ZC7w_pyCW8&maxResults=1&fields=items&textFormat=plainText&alt=json returned "The request specifies unexpected pagination parameters.". Details: "[{'message': 'The request specifies unexpected pagination parameters.', 'domain': 'youtube.parameter', 'reason': 'unexpectedParameter', 'location': 'maxResults,pageToken', 'locationType': 'parameter'}]">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\python\spam_base\YT-Spammer-Purge-2.16.4\YTSpammerPurge.py", line 1670, in <module>
    main()
  File "C:\python\spam_base\YT-Spammer-Purge-2.16.4\YTSpammerPurge.py", line 1647, in main
    continueRunning = primaryInstance(miscData)
  File "C:\python\spam_base\YT-Spammer-Purge-2.16.4\YTSpammerPurge.py", line 1599, in primaryInstance
    operations.check_deleted_comments(list(combinedCommentDict))
  File "C:\python\spam_base\YT-Spammer-Purge-2.16.4\Scripts\operations.py", line 1161, in check_deleted_comments
    unsuccessfulResults.append(results)
UnboundLocalError: local variable 'results' referenced before assignment
```

### Screenshots

![Screenshot 2022-03-08 124814](https://user-images.githubusercontent.com/95196459/157174496-5bb92881-7f6e-4ffb-a0a4-dcdbb60ce82b.png)

**Comment by TheAntVenom** [comment]
> Hm I ran into this myself once, I'll take a look

I just ran into this issue as well when trying to target all recent comments on my channel, inputting a specific channel id, and deleting them. I had the same 4 tracebacks as well.

**Comment by beaucarnes** [comment]
I get a similar error when deleting comments using multiple methods, with and without logging. It happens during the deleting process so some of the found spam comments are deleted and some do not get deleted before the error occurs. 

My workaround is to just run the program multiple times until it gets all the comments but it will be nice when this is fixed.

**Comment by ThioJoe** [comment]
Ok this should be fixed in 2.16.5 and https://github.com/ThioJoe/YT-Spammer-Purge/commit/fdcdd65d5bd3023a8843b5cdeb105bf57016ea7d . Turns out YouTube made an undocumented change to their API. The 'maxResults' parameter is not supposed to be used with the 'id' parameter, and I guess before it was just ignoring it, but now it returns an error.
